cmd/tnf/main.go

```diff
@@ -36,53 +38,97 @@ var (
 		Short: "adding new handler.",
 		RunE:  generateHandlerFiles,
 	}
+
+	defaultHandlersFolder = path.Join("pkg", "tnf", "handlers")
 )
 
-func generateHandlerFiles(cmd *cobra.Command, args []string) error {
-	handlername = args[0]
-	pathrelativetoroot = path.Join("..", "..")
-	handlerDirectory = path.Join(pathrelativetoroot, "pkg", "tnf", "handlers")
-	newHandlerDirectory := path.Join(handlerDirectory, handlername)
+func getHandlersDirectory() (string, error) {
+	handlersDirectory := os.Getenv(envHandlersFolder)
 
-	err := os.Mkdir(newHandlerDirectory, 0755)
-	if err != nil {
+	if handlersDirectory == "" {
+		log.Warnf("Environment variable %s not set. Handlers base folder will be set to ./%s",
+			envHandlersFolder, defaultHandlersFolder)
+
+		handlersDirectory = defaultHandlersFolder
+	} else {
+		log.Infof("Env var %s found. Handlers directory: %s", envHandlersFolder, handlersDirectory)
+	}
+
+	// Convert to absolute path.
+	if !path.IsAbs(handlersDirectory) {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return "", err
+		}
+
+		handlersDirectory = path.Join(cwd, handlersDirectory)
+	}
+
+	return handlersDirectory, nil
+}
+
+func generateHandlerFilesFromTemplates(handlerTemplatesDirectory, newHandlerDirectory string, myhandler myHandler) error {
+	templateFilePath := path.Join(handlerTemplatesDirectory, "doc.tmpl")
+	renderedFileName := docFileName
+
+	if err := createfile(templateFilePath, renderedFileName, myhandler, newHandlerDirectory); err != nil {
```

Comment:
can you add a log.Error here.
Also, I think this is worth a panic, we can't do anything if the files can't be created

Reply:
I've added an error log, but I don't think we need to panic here. If this function fails, the error is catched and the program returns immediately with a graceful exit.